src/app/core/services/basket/basket-payment.service.ts

```diff
@@ -70,42 +73,55 @@ export class BasketPaymentService {
         map(({ data, included }) =>
           data && data.paymentMethod && included ? included.paymentMethod[data.paymentMethod] : undefined
         ),
-        concatMap(pm => {
-          if (
-            !pm ||
-            !pm.capabilities ||
-            !pm.capabilities.some(data => ['RedirectBeforeCheckout'].includes(data)) // ToDo: add RedirectAfterCheckout here, if placeholders are supported by the ICM server
-          ) {
-            return of(paymentInstrument);
-            // send redirect urls if there is a redirect required
-          } else {
-            const redirect = {
-              successUrl: `${location.origin}/checkout/review?redirect=success`,
-              cancelUrl: `${location.origin}/checkout/payment?redirect=cancel`,
-              failureUrl: `${location.origin}/checkout/payment?redirect=failure`,
-            };
-
-            if (pm.capabilities.some(data => ['RedirectAfterCheckout'].includes(data))) {
-              redirect.successUrl = `${location.origin}/checkout/receipt?redirect=success&orderId=*orderID*`; // *OrderID* will be replaced by the ICM server
-              redirect.cancelUrl = `${location.origin}/checkout/payment?redirect=cancel&orderId=*orderID*`;
-              redirect.failureUrl = `${location.origin}/checkout/payment?redirect=failure&orderId=*orderID*`;
-            }
-
-            const body = {
-              paymentInstrument,
-              redirect,
-            };
-
-            return this.apiService
-              .put(`baskets/${basketId}/payments/open-tender`, body, {
-                headers: this.basketHeaders,
-              })
-              .pipe(mapTo(paymentInstrument));
-          }
-        })
+        withLatestFrom(this.store.pipe(select(getCurrentLocale))),
+        concatMap(([pm, currentLocale]) =>
+          this.sendRedirectUrlsIfRequired(pm, paymentInstrument, basketId, currentLocale)
+        )
       );
   }
 
+  /**
+   *  Checks, if RedirectUrls are requested by the server and sends them if it is necessary.
+   * @param pm                The payment method to determine if redirect is required.
+   * @param paymentInstrument The payment instrument id.
+   * @param basketId          The basket id.
+   * @param currentLocale     The current locale to compose the redirect urls.
+   * @returns                 The payment instrument id.
+   */
+  private sendRedirectUrlsIfRequired(
+    pm: PaymentMethodBaseData,
+    paymentInstrument: string,
+    basketId: string,
+    currentLocale: Locale
+  ): Observable<string> {
+    if (!pm || !pm.capabilities || !pm.capabilities.some(data => ['RedirectBeforeCheckout'].includes(data))) {
+      return of(paymentInstrument);
+      // send redirect urls if there is a redirect required
+    } else {
+      const redirect = {
+        successUrl: `${location.origin}/checkout/review;lang=${currentLocale.lang}?redirect=success`,
```

Comment:
helper variable for `location.origin` and `currentLocale.lang` would shorten these lines.